### ISSUE TITLE
Feature request: Customize the questions and flow in sam init

### ISSUE THREAD

**Issue by wagneraw** [aws/aws-sam-cli#4794]
### Describe your idea/feature/enhancement

It would be very useful to be able to customize the options presented to users when they run "sam init". This would allow us to have "sam init" present a list of customized getting started templates that are customized to our environment. This would provide an even more effective on-ramp for our developers.

### Proposal

I would think that it could be done similar to how you can customize the questions presented in the sam pipelines command, by providing a questions.json file or something similar that would change the behavior.

Things to consider:
1. Will this require any updates to the [SAM Spec](https://github.com/awslabs/serverless-application-model)

**Comment by sriram-mv** [comment]
This is interesting! I would like to dig deeper. Does this mean that you will be getting the applications from your private git repo?

`sam init` has `--location` today to load from a from a custom location/folder/zip etc.

**Comment by wagneraw** [comment]
Exactly, instead of developers needing to know all the different applications in the private git repo, they would be presented a text menu, just like in a normal "sam init" it would just be a menu of these private options.

**Comment by rwiturralde** [comment]
Same desire here. We'd like to maintain an internal (private) git repo where we build a set of cookiecutters for common patterns for our internal serverless apps. It would be ideal to be able to update the SAM CLI config to include this location and have our developers, using sam init, to be asked which example repo they want to navigate (the AWS SAM public and any additionally configured private repos) and, if they choose our private repo, we can control the subsequent questions they are asked.
